rnd/autogpt_server/autogpt_server/data/graph.py

```diff
@@ -290,11 +374,30 @@ async def create_graph(graph: Graph) -> Graph:
 
     await asyncio.gather(
         *[
-            AgentNode.prisma().create(
+            AgentGraph.prisma(tx).create(
```

Comment:
First persist the actual graph, then persist all of it's subgraph.
subgraph is simply a logical grouping, so there will be no nested subgraph, even if the graph is actually nested.